mlflow/java/client/README.md

```diff
@@ -0,0 +1,165 @@
+# MLflow Java Client
+
+Java client for [MLflow](https://mlflow.org) REST API.
+See also the MLflow [Python API](https://mlflow.org/docs/latest/python_api/index.html)
+and [REST API](https://mlflow.org/docs/latest/rest_api.html).
+
+## Requirements
+
+* Java 1.8
+* Maven
+* Run the [MLflow Tracking Server 0.4.2](https://mlflow.org/docs/latest/tracking.html#running-a-tracking-server)
+
+## Build
+
+### Build with tests
+
+The MLflow Java client tests require that MLflow is on the PATH (to start a local server),
+so it is recommended to run them from within a development conda environment.
+
+To build a deployable JAR and run tests:
+
+```
+mvn package
+```
+
+## Run
+
+To run a simple sample.
+```
+java -cp target/mlflow-java-client-0.4.2.jar \
+  com.databricks.mlflow.client.samples.QuickStartDriver http://localhost:5001
+```
+
+## JSON Serialization
+
+MLflow Java client uses [Protobuf](https://developers.google.com/protocol-buffers/) 3.6.0 to serialize the JSON payload.
+* [service.proto](../mlflow/protos/service.proto) - Protobuf definition of data objects.
+* [com.databricks.api.proto.mlflow.Service.java](src/main/java/com/databricks/api/proto/mlflow/Service.java) - Generated Java classes of all data objects.
+* [generate-protos.sh](generate-protos.sh) - One time script to generate Service.java. If service.proto changes you will need to re-run this script.
+* Javadoc can be generated by running ``mvn javadoc:javadoc``. The output will be in [target/site/apidocs/index.html](target/site/apidocs/index.html). 
+Here is the javadoc for [Service.java](target/site/apidocs/com/databricks/api/proto/mlflow/Service.html).
+
+## Java Client API
+
+See [ApiClient.java](src/main/java/org/mlflow/client/ApiClient.java) 
+and [Service.java domain objects](src/main/java/org/mlflow/api/proto/mlflow/Service.java).
+
+```
```

Comment:
Need to update some of the calls here to show the actual methods we settled on.

Reply:
Updated.